samples/image-uploader/src/index.js

```diff
@@ -0,0 +1,437 @@
+    }
+  }
+
+  onClickRemove = () => {
+    this.props.sdk.field.setValue(null)
+    this.setState({
+      value: null,
+      asset: null
+    })
+  }
+
+  onDragOverEnd = () => {
+    this.setState({ isDraggingOver: false })
+  }
+
+  onDragOverStart = () => {
+    this.setState({ isDraggingOver: true })
+  }
+
+  onError = error => {
+    console.error(error)
+    this.props.sdk.notifier.error(error.message)
+  }
+
+  onExternalChange = value => {
+    this.setState({ value })
+  }
+
+  /*
+     Create a new (unprocessed) asset entry for given upload and file.
+
+     createAsset(upload: UploadEntity, file: File, locale: string): Promise<AssetEntity>
+  */
+  createAsset = (upload, file, locale) => {
+    const asset = {
+      fields: {
+        title: {},
+        description: {},
+        file: {}
+      }
+    }
+
+    asset.fields.title[locale] = file.name
+    asset.fields.description[locale] = file.name
+    asset.fields.file[locale] = {
+      contentType: file.type,
+      fileName: file.name,
+      uploadFrom: {
+        sys: {
+          type: "Link",
+          linkType: "Upload",
+          id: upload.sys.id
+        }
+      }
+    }
+
+    return this.props.sdk.space.createAsset(asset)
+  }
+
+  // createAssetWithImageUrl(imageUrl, contentType, locale)
+  createAssetWithImageUrl = (imageUrl, contentType, locale) => {
+    const asset = {
+      fields: {
+        title: {},
+        description: {},
+        file: {}
+      }
+    }
+
+    asset.fields.title[locale] = imageUrl
+    asset.fields.description[locale] = imageUrl
+    asset.fields.file[locale] = {
+      contentType,
+      fileName: imageUrl,
+      upload: imageUrl
+    }
+
+    return this.props.sdk.space.createAsset(asset)
+  }
+
+  /*
+    If customers prefers localization of references, always return default locale.
+    If not, return current locale.
+  */
+  findProperLocale() {
+    if (this.props.sdk.entry.fields[this.props.sdk.field.id].type === "Link") {
+      return this.props.sdk.locales.default
+    }
+
+    return this.props.sdk.field.locale
+  }
+
+  // reuseExistingAsset(assetId)
+  reuseExistingAsset = async assetId => {
+    let asset
+
+    try {
+      asset = await this.props.sdk.space.getAsset(assetId)
+    } catch (err) {
+      this.onError(err)
+    }
+
+    this.setState({
+      asset
+    })
+
+    await this.props.sdk.field.setValue(
+      {
+        sys: {
+          type: "Link",
+          linkType: "Asset",
+          id: assetId
+        }
+      },
+      this.findProperLocale()
+    )
+  }
+
+  createNewAssetFromFiles = async files => {
+    // Only one image at a time is supported. In the future, we can accept set of images per locale ?
+    if (files.length > 1) {
+      return this.onError(new Error("Please drop only one image at a time"))
+    }
+
+    const file = files[0]
+    this.setState({ file })
+    this.setUploadProgress(0)
+
+    const [contentType, isImage] = getMimeTypeByPath(file.name)
+    if (!isImage) {
```

Comment:
I feel we have to support all asset types properly. Assets can change their type later. So if someone by accident replaces the image file within an asset with a pdf doc the image uploader right now will break pretty badly. Another example when this can happen is if there are already existing entries with existing references to non image assets before the image uploader was installed. I suggest we fully support all assets. However if we opt not to support all types of assets than we should at least not break the image uploader if it is confronted with non image assets.

Reply:
Fixed